Library/Homebrew/version.rb

```diff
@@ -379,3 +392,29 @@ def self._parse(spec)
     return m.captures.first unless m.nil?
   end
 end
+
+class HeadVersion < Version
+  attr_reader :commit
+
+  def initialize(val)
+    unless val.respond_to?(:to_str)
```

Comment:
Using the base class initializer is of course a good thing. :+1: I'm not overly bothered with the duplicate check, but I think we could fairly safely remove it in the constructors, as we no longer expect to call `[Head]Version.new` directly.